### PR TITLE
Update COMPONENTS for latest NCEPLIBS releases

### PR DESCRIPTION
**Pull request by kgerheiser**
GCC 10 fixes

Settable flags through CMAKE_<lang>_FLAGS

Remove -xHOST in some libraries to make them more portable

Clean-up compiler detection logic

### CODE REVIEW

File: COMPONENTS [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+g2          | v3.4.1            | 3.4.1
+g2tmpl      | v1.9.1            | 1.9.1
 crtm        | v2.3.0            | 2.3.0
 nceppost    | dceca26           | dceca26
-wrf_io      | v1.1.1-cmake      | 1.1.1
+wrf_io      | v1.1.1-cmake-v2      | 1.1.1
 bufr        | bufr_v11.3.1.1    | 11.3.1.1
 wgrib2      | v2.0.8-cmake-v1.0.3 | 2.0.8
---------------------------------------------------------
+--------------------------------------------------------
```
Comment (climbfuji) [inline COMPONENTS:23]: Is the newline required in line 23? Just checking.

**Review by climbfuji — APPROVED** [review]
I can't tell if the version numbers are all correct, but at least I see changes for almost all NCEPLIBS-* components.

File: COMPONENTS (hunk 1 above)
Comment (aerorahul) [reply COMPONENTS:23]: no. it is just a text file and we `grep` for package names.  It is purely decorative.

**Comment by aerorahul** [comment]
Can you please tag the latest bufr library too? 
I think develop is further along than the bufr tag here.

**Comment by kgerheiser** [comment]
Updated bufr

**Review by aerorahul — APPROVED** [review]
thanks!

**Pull request merged**